### ISSUE TITLE
Cactus-comments CORS and other issues

### ISSUE THREAD

**Issue by roughnecks** [spantaleev/matrix-docker-ansible-deploy#2106]
**Describe the bug**

Cannot get to load "cactus.js" because has been blocked by CORS policy.

I've explained the issue on gitlab upstream too, see [here](https://gitlab.com/cactus-comments/cactus-appservice/-/issues/11)

First of all, the `defaultHomeserverUrl` seems wrong as "https://matrix.woodpeckersnest.space:8448" is returning me an error message: 
`No Such Resource. No such child resource`. So I changed that to "https://woodpeckersnest.space", to no avail.

Second, the <script> tag in `<head>` must be changed in this way:

```
<script type="module" src="https://matrix.woodpeckersnest.space/cactus-comments/cactus.js"></script>
```

i.e, `type=module`, otherwise with `type="text/javascript"` Chrome web console complains that "Uncaught SyntaxError: Cannot use import statement outside a module (at cactus.js:1:1)"

One that is fixed (at least I think it's fixed), the CORS error appears, because js is on matrix.domain.tld, while website is on domain.tld.

```
Access to script at 'https://matrix.woodpeckersnest.space/cactus-comments/cactus.js' from origin 'https://woodpeckersnest.space' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**To Reproduce**

My `vars.yml` file looks like this:

```yaml
# Cactus Comments
matrix_cactus_comments_enabled: true

# Serve the base domain as well (unless you already have something serving it)
matrix_nginx_proxy_base_domain_serving_enabled: true
# Enable static webserver
matrix_nginx_proxy_base_domain_homepage_enabled: false
```

**Matrix Server:**
 - OS: Debian 11
 - Architecture: amd64

**Comment by spantaleev** [comment]
Both issues reported here have since been resolved. The playbook was fixed in January 2023 to download the compiled cactus.js and style.css from the official release, instead of serving the source version. Later, in January 2024, cactus-comments client serving was reworked into a dedicated matrix-cactus-comments-client role, which serves the files via a static-web-server container behind Traefik, so the old matrix-nginx-proxy setup that caused the CORS problem is gone entirely. Closing as fixed. If you still hit trouble with a current version of the playbook, please open a new issue.
